source/splitSingletrack/mod_splitSingletrack.f90

```diff
@@ -0,0 +1,210 @@
+module mod_splitSingletrack
+  use iso_fortran_env, only : output_unit
+  implicit none
+
+  ! File units used for I/O;
+  ! can be overridden USEing if clashing
+  ! with units used in rest of the code.
+  integer :: inUnit = 90
+  integer :: outUnit = 91
+
+  integer :: myOutUnit = output_unit
+
+  ! Temp arrays for I/O
+  character(len=80), private :: sixtit,commen
+  character(len=8),  private :: cdate,ctime,progrm
+  integer, private           :: i,ia,ia0,ifipa,ilapa,itopa,j,ntwin,numl,n
+  double precision, private  :: dpscor,sigcor
+  integer, private           ::  icode
+  double precision, private  :: b,c,c1,d,d1,dizu0,dmmac,dnms,dnumlr,dummy,e,e1,f,f1,g,g1,h,h1,p,p1
+  double precision, private  :: qwc(3)
+  double precision, private  :: clo(3)
+  double precision, private  :: clop(3)
+  double precision, private  :: di0(3)
+  double precision, private  :: dip0(3)
+  double precision, private  :: ta(6,6)
+
+contains
+  function numSTFpairs(ifname)
+    character(len=*), intent(in) :: ifname
+    integer numSTFpairs
+    integer iostat
+
+    open(unit=inUnit, file=ifname, form='UNFORMATTED', status='OLD', action='READ', err=510, iostat=iostat)
+
+    read(inUnit,end=511,err=520,iostat=iostat) &
+         sixtit,commen,cdate,ctime,       &
+         progrm,ifipa,ilapa,itopa,icode,numl,qwc(1),qwc(2),qwc(3), clo(1), &
+         clop(1),clo(2),clop(2),clo(3),clop(3), di0(1),dip0(1),di0(2),dip0(2), &
+         dummy,dummy, ta(1,1),ta(1,2),ta(1,3),ta(1,4),ta(1,5),ta(1,6), &
+         ta(2,1),ta(2,2),ta(2,3),ta(2,4),ta(2,5),ta(2,6), ta(3,1),ta(3,2), &
+         ta(3,3),ta(3,4),ta(3,5),ta(3,6), ta(4,1),ta(4,2),ta(4,3),ta(4,4), &
+         ta(4,5),ta(4,6), ta(5,1),ta(5,2),ta(5,3),ta(5,4),ta(5,5),ta(5,6), &
+         ta(6,1),ta(6,2),ta(6,3),ta(6,4),ta(6,5),ta(6,6), dmmac,dnms,dizu0,&
+         dnumlr,sigcor,dpscor
+    close(inUnit)
+
+    if (modulo(itopa,2) /= 0) then
```

Comment:
Should first check that `itopa` > 0. Also, mod and modulo behaves differently, but maybe not for integers.

Reply:
AFAIK the difference only shows up if one of the numbers are negative, which can't happen.

I added the check (which would really be a case of corrupted file).